common/src/main/java/com/linecorp/centraldogma/common/MergerQuery.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * A merger query on files.
+ *
+ * @param <T> the content type of files being queried
+ */
+public interface MergerQuery<T> {
```

Comment:
Google search shows me that 'merge query' is a fairly common term. How about renaming to `MergeQuery`?